### DISCUSSION TITLE
How to enable load_from_svg_data?

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#4162] [Q&A]
I'm developing an app on an embedded mcu (ESP32).

I'm trying to use `load_from_svg_data` (https://slint.dev/releases/1.3.2/docs/rust/slint/struct.Image#method.load_from_svg_data) but the compiler complains it's not available.
I see it's under "svg" feature, so I tried adding it to slint in `Cargo.toml`, but then it complains slint doesn't have such features, so maybe it is in some internal crate.

How can I use it and/or how can I load svg data into an Image from data in my app?

**Comment by Vadoola** [comment]
I use the load_from_svg_data just find in my application, however this is a desktop application. I've never done any development with ESP32, but I assume it would be a no_std environment, and that could be the problem?

Doing a very brief looking (since no one else has responded yet) It looks like Slint uses resvg for SVG support, and doing a quick glance at resvg I couldn't see any obvious sign that it support no_std (but I didn't look to deep).

So my **guess** would be that it's an issue with being a no_std environment. I'm sure someone who is a bit more knowledgeable will chime in when they get a chance. In the mean time perhaps this can point you in a useful direction?

**Reply by tronical** [reply]
Yes, indeed, SVG rendering is not supported in no_std environments yet. I've created https://github.com/slint-ui/slint/issues/4173 to track this, as it might very much make sense to support this in the long term.
